### PR TITLE
CTECH-1344: Fix tests for forked PR flow.

### PR DESCRIPTION
**Pull request by pms1969**
Signed-off-by: Paul Saunders <pms1969@gmail.com>

# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Ensures that the forked PR flow can execute and pass tests that it is able to run.
This has required:
* Skipping tests that rely on a secrets.json file
* Some minor changes to some tests to use the PersonalAccessToken where appropriate.

### CODE REVIEW

File: sdk/tests/tutorials/ibor/test_bitemporal.py [hunk 1]
```diff
@@ -35,14 +35,17 @@ def tearDownClass(cls):
...
     def print_transactions(self, as_at, transactions):
-        print("transactions at: {}".format(as_at))
-
-        for transaction in transactions:
-            print("{}\t{}\t{}\t{}\t{}".format(transaction.instrument_uid,
-                                              transaction.transaction_date,
-                                              transaction.units,
-                                              transaction.transaction_price.price,
-                                              transaction.total_consideration.amount))
+        # print("transactions at: {}".format(as_at))
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_bitemporal.py:38]: delete the commented code/function

File: sdk/tests/tutorials/ibor/test_properties.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Properties(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         # create a configured API client
-        api_client = ApiClientBuilder().build(CredentialsSource.secrets_path())
+        api_client = ApiClientBuilder().build()
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/ibor/test_properties.py:22]: was a need to do this? it please that locally devs will now need to use env vars rather that a secrets file

Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_properties.py:22]: OK, there were a couple of them.  I'll need to rethink how to do that then.  If you have the env vars, you want them to trump the secrets file if it even exists.

File: sdk/tests/tutorials/ibor/test_bitemporal.py (hunk 1 above)
Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_bitemporal.py:38]: ah, missed it.  Will take care of.

File: sdk/tests/tutorials/properties/test_derived_properties.py [hunk 3]
```diff
@@ -56,7 +56,8 @@ def create_ratings_property(self, *ratings):
                         f"Property {property_definition.domain}/{property_definition.scope}/{property_definition.display_name} already exists"
                     )
             finally:
-                self.id_generator.add_scope_and_code("property_definition", property_definition.scope, property_definition.code, ["Instrument"])
+                self.id_generator.add_scope_and_code("property_definition", property_definition.scope,
+                                                     property_definition.code, ["Instrument"])
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/properties/test_derived_properties.py:60]: is there some rogue formatting happening here?

File: sdk/tests/tutorials/properties/test_derived_properties.py [hunk 4]
```diff
@@ -133,7 +134,8 @@ def test_derived_property(self):
                     f"Property {derived_prop_definition_req.domain}{derived_prop_definition_req.scope}/{derived_prop_definition_req.display_name} already exists"
                 )
         finally:
-            self.id_generator.add_scope_and_code("property_definition", derived_prop_definition_req.scope, derived_prop_definition_req.code, ["Instrument"])
+            self.id_generator.add_scope_and_code("property_definition", derived_prop_definition_req.scope,
+                                                 derived_prop_definition_req.code, ["Instrument"])
```
Comment (rizwansaeed) [inline sdk/tests/tutorials/properties/test_derived_properties.py:138]: formatting?

File: sdk/tests/tutorials/ibor/test_bitemporal.py (hunk 1 above)
Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_bitemporal.py:38]: oh, no, I left that in there in case someone wanted it back.  I'm not a fan of print statements in tests or code as you know.  :|
Would it be best to revert the commenting or leave as is, or remove the stuff that I did comment, which may propagate to removing the method, and subsequently changing a number of tests???

File: sdk/tests/tutorials/properties/test_derived_properties.py (hunk 3 above)
Comment (pms1969) [reply sdk/tests/tutorials/properties/test_derived_properties.py:60]: yeah, turned on format on save.  It made a few changes in some places.

File: sdk/tests/tutorials/properties/test_derived_properties.py (hunk 4 above)
Comment (pms1969) [reply sdk/tests/tutorials/properties/test_derived_properties.py:138]: ja

File: sdk/tests/tutorials/ibor/test_bitemporal.py (hunk 1 above)
Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_bitemporal.py:38]: Gone

File: sdk/tests/tutorials/properties/test_derived_properties.py (hunk 3 above)
Comment (pms1969) [reply sdk/tests/tutorials/properties/test_derived_properties.py]: fixed

File: sdk/tests/tutorials/ibor/test_properties.py (hunk 2 above)
Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_properties.py]: sorted.

File: sdk/tests/tutorials/ibor/test_bitemporal.py [hunk 5]
```diff
@@ -34,16 +32,6 @@ def setUpClass(cls):
...
-    def print_transactions(self, as_at, transactions):
-        print("transactions at: {}".format(as_at))
-
-        for transaction in transactions:
-            print("{}\t{}\t{}\t{}\t{}".format(transaction.instrument_uid,
-                                              transaction.transaction_date,
-                                              transaction.units,
-                                              transaction.transaction_price.price,
-                                              transaction.total_consideration.amount))
-
```
Comment (MikeMcGarry) [inline sdk/tests/tutorials/ibor/test_bitemporal.py:46]: Is this just because you didn't like the fact that we are somewhat randomly printing these?

File: sdk/lusid/utilities/api_client_builder.py [hunk 6]
```diff
@@ -65,8 +65,8 @@ def resolve_api_token(cls, token=None, api_secrets_filename=None, configuration=
             cls.__check_required_fields(configuration, ["api_url"])
             api_token = token
 
-        # If there is a token in the env vars, and the user has not provided a secrets file, use the token
-        elif configuration.access_token is not None and api_secrets_filename is None:
+        # If there is a token provided use it
+        elif configuration.access_token is not None:
```
[the remark this answers is not in the document]
Comment (rizwansaeed) [reply sdk/lusid/utilities/api_client_builder.py:69]: we need to be mindful there are 2 different precedence ordering:
1) env vars vs secrets
2) access token vs resource owner flow

File: sdk/tests/utilities/test_api_client_builder.py [hunk 7]
```diff
@@ -40,15 +39,16 @@ def test_missing_from_config_file_throws(self, _, missing_attributes, token):
         proxy_config = ProxyConfig(**{
             key.replace("proxy_", ""): value for key, value in source_config_details.items() if
             value is not None and "proxy" in key
-        }) if source_config_details["proxy_address"] is not None else None
+        }) if source_config_details.get("proxy_address", None) is not None else None
 
-        api_config_kwargs = {key: value for key, value in source_config_details.items() if
-                             value is not None and "proxy" not in key}
+        api_config_kwargs = {key: "value" for key, value in config_keys.items() if "proxy" not in key}
```
Comment (MikeMcGarry) [inline sdk/tests/utilities/test_api_client_builder.py:44]: Why did you remove if `value is not None`?

File: sdk/tests/utilities/test_api_client_builder.py [hunk 8]
```diff
@@ -59,6 +59,7 @@ def test_missing_from_config_file_throws(self, _, missing_attributes, token):
                                   f"please ensure that you have provided them directly, via a secrets file or environment "
                                   f"variables")
 
+    @unittest.skipIf(CredentialsSource.fetch_credentials().__contains__("access_token"), "do not run on PR's")
```
Comment (MikeMcGarry) [inline sdk/tests/utilities/test_api_client_builder.py:62]: Ahhh, nice

File: sdk/tests/tutorials/ibor/test_bitemporal.py (hunk 5 above)
Comment (pms1969) [reply sdk/tests/tutorials/ibor/test_bitemporal.py:46]: You missed my codecraft. :)   Prints and log output in tests is a really bad idea.  There's the potential to leak sensitive information to a medium that isn't sensitive to it, and if you aren't careful, something you are using for debug purposes is now spamming 1000's of lines to stdout/stderr, which can significantly hinder the task of identifying what is wrong overall.

File: sdk/tests/tutorials/marketdata/test_corporate_actions.py [hunk 9]
```diff
@@ -19,7 +19,7 @@ class CorporateActions(unittest.TestCase):
     @classmethod
     def setUp(cls):
         # create a configured API client
-        api_client = ApiClientBuilder().build(str(CredentialsSource.secrets_path()))
+        api_client = ApiClientBuilder().build(CredentialsSource.secrets_path())
```
[the remark this answers is not in the document]
Comment (pms1969) [reply sdk/tests/tutorials/marketdata/test_corporate_actions.py:22]: it is. That call `ApiClientBuilder().build(CredentialsSource.secrets_path())` is used in any number of other places without the cast to `str` so it's a consistency thing if nothing else.

File: sdk/tests/utilities/test_api_client_builder.py [hunk 10]
```diff
@@ -40,15 +39,16 @@ def test_missing_from_config_file_throws(self, _, missing_attributes, token):
         proxy_config = ProxyConfig(**{
             key.replace("proxy_", ""): value for key, value in source_config_details.items() if
             value is not None and "proxy" in key
-        }) if source_config_details["proxy_address"] is not None else None
+        }) if source_config_details.get("proxy_address", None) is not None else None
```
[the remark this answers is not in the document]
Comment (pms1969) [reply sdk/tests/utilities/test_api_client_builder.py:42]: it did in my testing at one point, although I appreciate it's been working forever.

File: sdk/tests/utilities/test_api_client_builder.py (hunk 8 above)
Comment (pms1969) [reply sdk/tests/utilities/test_api_client_builder.py:62]: clever huh... :)

File: sdk/tests/utilities/test_api_client_builder.py (hunk 7 above)
Comment (pms1969) [reply sdk/tests/utilities/test_api_client_builder.py:44]: it changed from `source_config_details.items()` to `config_keys.items()` as the source of the key/values.  There are no values in `config_keys`.

File: sdk/tests/tutorials/ibor/test_bitemporal.py (hunk 5 above)
Comment (MikeMcGarry) [reply sdk/tests/tutorials/ibor/test_bitemporal.py:46]: Yeah of course, makes sense to me, thanks for doing this.

File: sdk/tests/tutorials/marketdata/test_corporate_actions.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply sdk/tests/tutorials/marketdata/test_corporate_actions.py:22]: Okay nice spot.

File: sdk/tests/utilities/test_api_client_builder.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply sdk/tests/utilities/test_api_client_builder.py:42]: Yeah we must populate it with an empty key for "proxy_address", this is much better handling, thanks!

File: sdk/tests/utilities/test_api_client_builder.py (hunk 7 above)
Comment (MikeMcGarry) [reply sdk/tests/utilities/test_api_client_builder.py:44]: Roger.

File: sdk/lusid/utilities/api_client_builder.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply sdk/lusid/utilities/api_client_builder.py:69]: @pms1969 Can we please remove the `api_secrets_filename` parameter from this function as it is not used anymore.

File: sdk/Dockerfile [hunk 11]
```diff
@@ -10,8 +10,5 @@ RUN pip install --no-cache-dir -r requirements.txt -r requirements.dev.txt
 
 RUN pip --no-cache-dir install --upgrade awscli
 
-ENV FBN_LUSID_API_URL ${FBN_LUSID_API_URL}
-ENV FBN_LUSID_ACCESS_TOKEN ${FBN_LUSID_ACCESS_TOKEN}
```
[the remark this answers is not in the document]
Comment (MikeMcGarry) [reply sdk/Dockerfile:14]: I think the intention was to provide these at runtime, but as you say this isn't the right way to do that.

**Review by MikeMcGarry — APPROVED** [review]
Looks great, nice work!

File: sdk/lusid/utilities/api_client_builder.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (pms1969) [reply sdk/lusid/utilities/api_client_builder.py:67]: done
